src/main/java/com/three_tech_solutions/slot_app/services/implementations/SlotServiceImpl.java

```diff
@@ -37,6 +46,28 @@ public void createSlot(CreateSlotRequest request) {
         slotRepository.save(buildSlot(request));
     }
 
+    @Override
+    public ListSlotsResponse getSlotsByDayOfWeek(User user, DayOfWeek dayOfWeek) {
+        return getSlotsByUserAndDayOfWeek(user, dayOfWeek).stream()
+                .map(slot -> slotMapper.toSlotResponse(slot, calculateUsedCapacity(slot)))
+                .collect(collectListAndBuildListSlotsResponse());
+    }
+
+    private List<Slot> getSlotsByUserAndDayOfWeek(User user, DayOfWeek dayOfWeek) {
+        return slotRepository.findAllByUserIdAndDayOfWeekOrdered(user, dayOfWeek);
+    }
+    private Collector<SlotResponse, Object, ListSlotsResponse> collectListAndBuildListSlotsResponse() {
+        return Collectors.collectingAndThen(
+                Collectors.toList(),
+                list -> new ListSlotsResponse(list.size(), list)
+        );
+    }
+
+    private int calculateUsedCapacity(Slot slot) {
+        if (slot.getStudents() == null) return 0;
+        return slot.getStudents().size();
```

Comment:
Esta validacion no te haria falta si hicieramos que la lista de estudiantes de un turno por defecto se setee como una lista vacia


```suggestion
        return slot.getStudents().size();
```